tests/std/tests/P0896R4_ranges_alg_permutations/test.cpp

```diff
@@ -0,0 +1,188 @@
+                assert(result.found);
+                assert(equal(input, Expected[i]));
+            }
+
+            R range{input};
+            const same_as<next_permutation_result<iterator_t<R>>> auto result =
+                next_permutation(range.begin(), range.end(), ranges::less{}, get_val);
+            assert(result.in == range.end());
+            assert(!result.found);
+            assert(equal(input, Expected[0]));
+        }
+    }
+};
+
+template <auto& Expected>
+struct prev_perm_instantiator {
+    template <ranges::bidirectional_range R>
+    static constexpr void call() {
+        using ranges::prev_permutation, ranges::prev_permutation_result, ranges::equal, ranges::is_sorted,
+            ranges::iterator_t;
+
+        constexpr auto count = static_cast<int>(ranges::size(Expected));
+
+        { // Validate range overload
+            int_wrapper input[ranges::size(Expected[0])];
+            ranges::copy(Expected[count - 1], input);
+
+            for (int i = count - 1; i-- > 0;) {
+                R range{input};
+                const same_as<prev_permutation_result<iterator_t<R>>> auto result =
+                    prev_permutation(range, ranges::less{}, get_val);
+                assert(result.in == range.end());
+                assert(result.found);
+                assert(equal(input, Expected[i]));
+            }
+
+            R range{input};
+            const same_as<prev_permutation_result<iterator_t<R>>> auto result =
+                prev_permutation(range, ranges::less{}, get_val);
+            assert(result.in == range.end());
+            assert(!result.found);
+            assert(equal(input, Expected[count - 1]));
+        }
+
+        { // Validate iterator overload
+            int_wrapper input[ranges::size(Expected[0])];
+            ranges::copy(Expected[count - 1], input);
+
+            for (int i = count - 1; i-- > 0;) {
+                R range{input};
+                const same_as<prev_permutation_result<iterator_t<R>>> auto result =
+                    prev_permutation(range.begin(), range.end(), ranges::less{}, get_val);
+                assert(result.in == range.end());
+                assert(result.found);
+                assert(equal(input, Expected[i]));
+            }
+
+            R range{input};
+            const same_as<prev_permutation_result<iterator_t<R>>> auto result =
+                prev_permutation(range.begin(), range.end(), ranges::less{}, get_val);
+            assert(result.in == range.end());
+            assert(!result.found);
+            assert(equal(input, Expected[count - 1]));
+        }
+    }
+};
+
+int main() {
+#if defined(__clang__) || defined(__EDG__) // TRANSITION, VSO-938163
+    STATIC_ASSERT((test_bidi<next_perm_instantiator<perm4>, int_wrapper>(), true));
+#endif // TRANSITION, VSO-938163
+    test_bidi<next_perm_instantiator<perm4>, int_wrapper>();
+    test_bidi<next_perm_instantiator<perm5>, int_wrapper>();

```

Comment:
Missing test coverage for
- Some (groups of) elements equal
- All elements equal

Reply:
Done. I added three "all elements equal" cases to get coverage for 0, 1, and at least one other size (8).